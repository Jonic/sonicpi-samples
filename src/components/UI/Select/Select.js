import PropTypes from 'prop-types'
import React from 'react'

import './Select.css'

const Select = (props) => {
  const {
    options,
    onChange,
    value,
    disabled,
  } = props

  return (
    <select
      className="c-Select"
      disabled={disabled}
      value={value}
      onChange={event => onChange(event.target.value)}
    >
      {
        options.map((option, index) => {
          return (
            <option key={index} value={option}>{option}</option>
          )
        })
      }
    </select>
  )
}

Select.propTypes = {
  disabled: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  options:  PropTypes.array.isRequired,
  value:    PropTypes.string.isRequired,
}

export default Select
