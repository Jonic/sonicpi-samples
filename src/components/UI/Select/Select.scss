@import '../../../styles/component-helper';

.c-select {}
