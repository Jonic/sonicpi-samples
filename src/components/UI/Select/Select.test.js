import React from 'react'
import { shallow } from 'enzyme'
import Select from './Select'

const options = []
const onChange = jest.fn()
const value = ''
const disabled = false

let wrapper

describe('<Select />', () => {
  beforeEach(() => {
    wrapper = shallow(
      <Select
        options={options}
        onChange={onChange}
        value={value}
        disabled={disabled}
      />
    )
  })

  describe('Renders...', () => {
    it('without crashing', () => {
      expect(wrapper.find('.c-select')).toBePresent()
    })

    it('matches its snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
