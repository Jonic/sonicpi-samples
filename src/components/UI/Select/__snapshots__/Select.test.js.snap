// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> Renders... matches its snapshot 1`] = `
<select
  className="c-Select"
  disabled={false}
  onChange={[Function]}
  value=""
/>
`;
